runelite-client/src/main/java/net/runelite/client/plugins/ibansstaff/IbansStaffPlugin.java

```diff
@@ -0,0 +1,230 @@
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.api.events.AnimationChanged;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Iban's Staff"
+)
+@Slf4j
+public class IbansStaffPlugin extends Plugin
+{
+	private static final Pattern chargesString = Pattern.compile("You have ([0-9]+) charges left on the staff.");
+	private static final String rechargeString = "...And feel the power of Zamorak flow through you.";
+	private static final int WEAPON_SLOT = 3;
+
+	private IbansStaffCounter counter;
+	private BufferedImage counterImage;
+	private int rechargeAmount;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private IbansStaffConfig config;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Getter
+	private int staffCharges;
+
+	@Provides
+	IbansStaffConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(IbansStaffConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		counterImage = itemManager.getImage(ItemID.IBANS_STAFF_U);
+		staffCharges = config.charges();
+		if (config.alwaysShow() || hasIbansEquipped())
+		{
+			addChargesCounter();
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		removeChargesCounter();
+	}
+
+	private void addChargesCounter()
+	{
+		if (counter == null)
+		{
+			counter = new IbansStaffCounter(counterImage, this);
+			infoBoxManager.addInfoBox(counter);
+			log.debug("Added Iban's Staff charges Counter");
+		}
+	}
+
+	private void removeChargesCounter()
+	{
+		if (counter != null)
+		{
+			infoBoxManager.removeInfoBox(counter);
+			counter = null;
+			log.debug("Removed Iban's Staff charges Counter");
+		}
+	}
+
+	private boolean hasIbansEquipped()
+	{
+		ItemContainer equipContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+		if (equipContainer != null)
+		{
+			Item[] equip = equipContainer.getItems();
+			if (equip.length >= WEAPON_SLOT)
+			{
+				int weaponId = equip[WEAPON_SLOT].getId();
+				return  weaponId == ItemID.IBANS_STAFF ||
+						weaponId == ItemID.IBANS_STAFF_U ||
+						weaponId == ItemID.IBANS_STAFF_1410;
+			}
+		}
+
+		return false;
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged event)
+	{
+		if (config.alwaysShow())
+		{
+			addChargesCounter();
+		}
+		else if (!hasIbansEquipped())
+		{
+			removeChargesCounter();
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		if (!config.alwaysShow())
```

Comment:
You should probably check if the event.getType is either inventory or equipment, because this will be quite a few checks everytime you open a bank.